076-support-for-kafka-4.0.md

```diff
@@ -0,0 +1,202 @@
+This is not a Strimzi's choice, but a result of the development in the Apache Kafka project.
+
+## Rejected alternatives
+
+### Skip the Strimzi 0.4y release
+
+We could consider skipping the Strimzi 0.4y release and instead remove the support for ZooKeeper-based clusters right after releasing Strimzi 0.4x.
+In that case, the Strimzi 0.4x will receive the extended support.
+But assuming the interval between Kafka 3.9 and 4.0 is 3-4 months, it seems that keeping the 0.4y release can help us implemented some more features and possibly fix some more bug fixes before we drop ZooKeeper support.
+Doing the 0.4y release would also minimize the risk of Apache Kafka introducing 3.10 release only later after the 3.9 release is done.
+This should also simplify the extended support, since we will have to do it for a shorter time in a separate release branch and safe some effort (from the 1 year of support window of Kafka 3.9, 1-2 months will be gone before we release the 0.4y with extended support).
+
+### Transition from Apache Kafka 3.9 directly to 4.0
+
+An alternative option would be to move from Kafka 3.9 directly to Kafka 4.0.
+That would mean that we would have one or more Strimzi version(s) with support for Kafka 3.8.x and 3.9.x.
+And once Kafka 4.0.0 will be released, we will drop support for 3.X completely and the next Strimzi version would support Kafka 4.0.x only.
+
+This approach was rejected, because:
+* It does not seem to offer support for 2 last minor Kafka versions which is something Strimzi offered for a long time.
+* We would either need to wait for Kafka 4.0.0 to be released (3-4 months after 3.9 release) to prepare the release, or we would anyway need to do all the work already with Kafka 3.9 anyway.
+
+So it would leave us possibly for a very long time without any release and does not seem to bring any real advantages over the approach suggested by this proposal.
+
+### Continue ZooKeeper support until Kafka 4.1 is released
+
+Another alternative option would be to continue releasing Strimzi versions based on Apache Kafka 3.9 and 4.0 until Kafka 4.1 is released while supporting ZooKeeper with Kafka 3.9.
+Once Kafka 4.1 would be released, we would drop support for Kafka 3.9 and for ZooKeeper and continue with Strimzi versions supporting Kafka 4.0 and 4.1 with KRaft only.
+
+The disadvantage of this approach is that it would require additional effort to validate the used versions and make sure that users are not trying to deploy the ZooKeeper based Kafka cluster with Kafka 4.0.
+It might also lead to confusing user experience.
+For example, a Kafka CR without explicitly specified Kafka version should use the latest supported Kafka version.
+Would that mean that ZooKeeper based clusters without specified `.spec.kafka.version` would automatically end up in error state because they would use Kafka 4.0?
+Or will we add some special handling that in such cases the version would default to Kafka 3.9?
```

Comment:
I think this should be easily handled by some boolean or? If ZK is used then use only Kafka 3.9, and if someone wants to deploy with a different version i.e., 4.0+ then throw an error.

Reply:
I do not think it would be a simple boolean.